### PR TITLE
Emissary- Ingress Security self assessment  and Thread Model

### PR DESCRIPTION
**Pull request by Disha-S-Gowda**
Hello emissary team and CNCF, 

We are a group of students from NYU, doing a security self-assessment under CNCF/tag-security. we have tried to understand emissary-ingress, its working and came up with a self-assessment . it will greatly aid us, if you can review it, and provide feedback/suggestions. 

Thanks in Advance!!.

**Contributors**:
@Disha-S-Gowda
@yashaswi2000
@jcart657
@Saipv17

### CODE REVIEW

File: assessments/projects/emissary-ingress/emissary-ingress-security-self-assessment.md [hunk 1]
```diff
@@ -0,0 +1,475 @@
+# Self-assessment
```
Comment (eddie-knight) [inline assessments/projects/emissary-ingress/emissary-ingress-security-self-assessment.md:1]:
```suggestion
# Emissary Ingress Self-assessment
```

File: assessments/projects/emissary-ingress/emissary-ingress-security-self-assessment.md
[the remark this answers is not in the document]
Comment (yashaswi2000) [reply assessments/projects/emissary-ingress/emissary-ingress-security-self-assessment.md:1]: changing it

**Comment by yashaswi2000** [comment]
Hi @eddie-knight, thanks for your initial review. we have cleaned up the commit history and removed the template text as you pointed out, name change to the file. Please let me know if you have any more comments in the future. Thanks!

File: assessments/projects/emissary-ingress/self-assessment.md [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+
+## Security issue resolution
+
+Related security flaws may be reported over the Slack channel or by email to the emissary ingress team. The group responds expeditiously.
+In addition, users may visit [status.getambassador](https://status.getambassador.io/) where any problems that arise while the active products (Emissary, Edge Stack, Telepresence, Ambassador Cloud) are being fixed will be reported.
+
+**Responsible Disclosures Process**
+
+Assisting with the vulnerability fix, the emissary ingress works with The Common Vulnerability Scoring System (also known as CVSS Scores). In Detail, As soon as the team is able to put up the fix, a patched version will be made available for serious severity problems .The issues with Medium CVSS score will be fixed in the next version, which usually happens in 4-6 weeks or less. In a later version, they address Low
```
Comment (ragashreeshekar) [inline assessments/projects/emissary-ingress/self-assessment.md:184]: This section is meant to show 
- Responsible disclosure - how and where an individual can report security vulnerabilities to the project. 
- Incident response - how does the project handle these reports incl. triaging, communicating to stakeholders, releasing the fix.

The content included seems to be the remediation communication. Please revisit to include relevant content. This is a good example for your reference - https://github.com/cncf/tag-security/blob/main/assessments/projects/pixie/self-assessment.md#security-issue-resolution. Kindly add references to the content if any.

File: assessments/projects/emissary-ingress/self-assessment.md [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+  - [2.2.1](https://www.getambassador.io/docs/emissary/latest/about/known-issues#221) TLS certificates using elliptic curves were incorrectly flagged as invalid. This issue is corrected in Emissary-ingress 2.2.2.
+  - [security compliance issues](https://github.com/emissary-ingress/emissary/issues?q=is%3Aissue+label%3Asecurity%2Fcompliance+is%3Aopen) lists the active issues in security and compliance.
+* [CII Best Practices](https://www.bestpractices.dev/en/projects).
+  Emissary-Ingress is in progress with 94% in Open Source Security Foundation (OpenSSF) best practices. [![OpenSSF Best Practices](https://www.bestpractices.dev/projects/1852/badge)](https://www.bestpractices.dev/projects/1852).
+  - unknown status of dynamic code analysis as: It is SUGGESTED that at least one dynamic analysis tool be applied to any proposed major production release of the software before its release.
+  - unknown status of The process for reporting vulnerabilities on the project site. (missing URL)
+  - If private vulnerability reports are supported, the project MUST include how to send the information in a way that is kept private. (missing URL)
```
Comment (ragashreeshekar) [inline assessments/projects/emissary-ingress/self-assessment.md:203]: Please include them under recommendations to the project or action items sub-section

**Review by ragashreeshekar — CHANGES_REQUESTED** [review]
Thanks for the PR @Disha-S-Gowda and team, appreciate the efforts.
I have completed first pass of review. Please feel free to reach out here or on slack for any questions and clarifications.

Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

File: assessments/projects/emissary-ingress/self-assessment.md (hunk 2 above)
Comment (Disha-S-Gowda) [reply assessments/projects/emissary-ingress/self-assessment.md]: We have modified the incident response and responsible disclosure based on your suggestion.

**Comment by yashaswi2000** [comment]
Hi @ragashreeshekar, thanks for your review. we addressed the changes suggested by you. please let us know any further comments or suggestions you have. also, it's not showing to me that the base branch is out of sync, and showing "Able to merge" so a bit confused as to what you are referring to.

File: assessments/projects/emissary-ingress/self-assessment.md [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+* Maintaining resiliance and performance as a API gateway.
+
+#### Security Goals 
+* Safeguard the declarative settings to prevent any unauthorized entity from altering them, which might potentially disrupt the routing pathways of the service. 
+* Make sure that the communication between Emissary-ingress components and other services is secured using robust protocols, such as Transport Layer Security (TLS). 
+* Supply logs and auditing trails for activities and events occurring inside Emissary-ingress to facilitate security monitoring and incident response. 
+* Implement robust authentication mechanisms to prevent unauthorized login to the cluster. 
+* Assuring that standard settings adhere to security best practices and do not expose superfluous services or functionality. 
+
+### Non-goals
```
[the remark this answers is not in the document]
Comment (yashaswi2000) [reply assessments/projects/emissary-ingress/self-assessment.md:107]: added functional non-goals as well.

File: assessments/projects/emissary-ingress/self-assessment.md [hunk 5]
```diff
@@ -0,0 +1,468 @@
+# Emissary Ingress Self-assessment
+
+Authors: Yashaswi Makula (@yashaswi2000), Disha Sheshappa (@Disha-S-Gowda), Sai Prasanth Virigineni (@Saipv17), Jarred Carter (@jcart657).
```
Comment (torinvdb) [inline assessments/projects/emissary-ingress/self-assessment.md:3]:
```suggestion
# Emissary Ingress Self-assessment

This assessment was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process and is currently pending changes from the maintainer team.

Authors: Yashaswi Makula (@yashaswi2000), Disha Sheshappa (@Disha-S-Gowda), Sai Prasanth Virigineni (@Saipv17), Jarred Carter (@jcart657).
```
@ragashreeshekar

**Review by ragashreeshekar — APPROVED** [review]

File: assessments/projects/emissary-ingress/self-assessment.md [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+- Emissary-ingress could not natively support third user authentication. 
+- Emissary-ingress primarily concerns itself with the administration of incoming external traffic at the boundary and may not have direct authority over or ensure the security of internal network connections. 
+- Emissary-ingress does not provide encryption key management for apps. 
+- Emissary-ingress does not focus on Certificate configuration/management and its recommended to use [Cert-managers](https://www.getambassador.io/docs/edge-stack/latest/howtos/cert-manager#using-cert-manager)
+
+
+## Self-Assessment Use
+
+This self-assessment is created by the emissary-ingress team to perform an internal analysis of the
+project's security.  It is not intended to provide a security audit of emissary-ingress, or
```
Comment (ragashreeshekar) [inline assessments/projects/emissary-ingress/self-assessment.md:135]:
```suggestion
This self-assessment is not intended to provide a security audit of emissary-ingress, or
```

**Review by JustinCappos — APPROVED** [review]
LGTM.  Appropriate level of detail and a good starting point.

**Pull request merged**